jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Node.java

```diff
@@ -77,14 +77,20 @@ public void insert(ForceObject<T> element) {
       return;
     }
     if (isLeaf()) {
-      // there already is a forceObject, so split
-      split();
-      // put the current resident and the new one into the correct quardrants
-      insertForceObject(this.forceObject);
-      insertForceObject(element);
-      // update the centerOfMass, Mass, and Force on this node
-      this.forceObject = this.forceObject.add(element);
-
+      if (this.forceObject.p.equals(element.p)) {
+        // compare points for special case where the 2 elements are at the same location
+        // this would cause an infinite attempt to split and re-insert
+        // just add the new mass
+        this.forceObject = this.forceObject.add(element);
+      } else {
+        // there already is a forceObject and location is different, so split
+        split();
+        // put the current resident and the new one into the correct quardrants
+        insertForceObject(this.forceObject);
+        insertForceObject(element);
+        // update the centerOfMass, Mass, and Force on this node
+        this.forceObject = this.forceObject.add(element);
+      }
     } else {
```

Comment:
prevents infinite recursion when a force object is inserted that is at the same location as a previous one.